### PR TITLE
fix(classes, core): add ability to map Date[] types

### PR DESCRIPTION
**Pull request by roblopz**
- Adding support for Date[] type mappings on classes/pojo.
- Adding tests for the above cases.

### CODE REVIEW

File: packages/core/src/lib/map/map.ts [hunk 1]
```diff
@@ -245,7 +245,7 @@ Original error: ${originalError}`;
       if (Array.isArray(mapInitializedValue)) {
         const [first] = mapInitializedValue;
         // if first item is a primitive
-        if (typeof first !== 'object') {
```
Comment (nartc) [inline packages/core/src/lib/map/map.ts:248]: praise: nice catch!

**Review by nartc — APPROVED** [review]

File: packages/classes/src/lib/utils/instantiate.util.ts [hunk 2]
```diff
@@ -69,7 +69,9 @@ export function instantiate<TModel extends Dictionary<TModel>>(
     // if is Date, assign a new Date value if valueAtKey is defined, otherwise, undefined
     if (isDateConstructor(metaResult)) {
       const value = isDefined(valueAtKey)
-        ? new Date(valueAtKey as number)
```
Comment (nartc) [inline packages/classes/src/lib/utils/instantiate.util.ts:72]: issue: we need to copy this same logic to `pojos/instantiate.util.ts` as well

**Review by nartc — COMMENTED** [review]
Good job! Just one small commetn

File: packages/classes/src/lib/utils/instantiate.util.ts (hunk 2 above)
Comment (roblopz) [reply packages/classes/src/lib/utils/instantiate.util.ts:72]: Added to pojos.

**Comment by nartc** [comment]
thank you for the pr

**Pull request merged**